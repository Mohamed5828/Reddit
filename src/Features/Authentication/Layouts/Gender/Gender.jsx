/* eslint-disable no-unused-vars */
/* eslint-disable react-hooks/exhaustive-deps */

import { SiReddit } from "react-icons/si";

import {
  AuthContainer,
  Privacy,
  AuthParagraph,
  AuthHeader,
  RedditIcon,
  Upper,
  Skip,
} from "./Gender.styled";

import GenderButton from "Features/Authentication/Components/GenderButton/GenderButton";

/**
 *  Gender component that is used in AfterSignUp Component
 * @param {Function} setGender Function to set the gender of the user
 * @param {Function} setGenderScreen Function to set the state of genderScreen
 * @param {Function} setInterestsScreen Function to set the state of interestsScreen
 * @returns {React.Component}  Gender component that is used in AfterSignUp Component
 */

const Gender = ({
  setGender,
  setGenderScreen,
  setInterestsScreen,
  setPhotoUploadScreen,
}) => {
  /**
   * Function to set the gender of the user and go to the next screen
   * @param {String} gender The Gender of the user
   */
  const nextScreen = (gender) => {
    //console.log(gender);
    setGender(gender);
    setGenderScreen(false);
    setPhotoUploadScreen(false);
    setInterestsScreen(true);
  };

  return (
    <>
      {
        <AuthContainer>
          <Upper>
            <RedditIcon>
              <span>
                <SiReddit size={60} color={"red"}></SiReddit>
              </span>
            </RedditIcon>
            <Skip
              onClick={() => {
                setGenderScreen(false);
                setInterestsScreen(true);
              }}
            >
              Skip
            </Skip>
          </Upper>
          <AuthHeader>About you</AuthHeader>
          <AuthParagraph>
            Tell us about yourself to start building your home feed.
          </AuthParagraph>
          <AuthParagraph identity={true}>How do you identify?</AuthParagraph>
          <GenderButton onClick={() => nextScreen("woman")}>Woman</GenderButton>
          <GenderButton onClick={() => nextScreen("man")}>Man</GenderButton>
        </AuthContainer>
      }
    </>
  );
};

export default Gender;
