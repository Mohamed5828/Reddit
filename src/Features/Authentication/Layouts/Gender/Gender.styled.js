import styled from "styled-components";

const fontFamily = "'IBM Plex Sans', sans-serif";

export const AuthContainer = styled.div`
  display: flex;
  flex-direction: column;

  max-width: 95%;
  margin: 0px auto 0;
  position: relative;
  width: 95%;
`;

export const AuthHeader = styled.h2`
  font-family: ${fontFamily};
  line-height: 24px;
  max-width: 280px;
  text-transform: none;
  display: flex;
  justify-content: center;
  font-weight: bold;
  font-size: 24px;
  margin: 10px 0;
  transform: translateX(22px);
`;
export const AuthParagraph = styled.div`
  color: ${({ theme }) => theme.color.authParagraphColor};
  font-family: Noto Sans, sans-serif;
  font-size: ${(p) => (p.identity ? "14px" : "12px")};
  font-weight: ${(p) => (p.identity ? "550" : "400")};
  line-height: 16px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 10px 0;
`;

export const Privacy = styled.span`
  color: ${({ theme }) => theme.color.link};
`;

export const RedditIcon = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;

  svg {
    transform: translateX(18px);
  }
`;

export const Upper = styled.div`
  display: flex;
  justify-content: center;

  span {
    cursor: pointer;
    font-size: 16px;
    font-weight: 600;
  }
`;

export const Skip = styled.span``;
