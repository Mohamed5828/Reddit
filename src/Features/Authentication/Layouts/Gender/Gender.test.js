import { shallow } from "enzyme";

import Gender from "./Gender";

describe("Gender Component", () => {
  it("this is a test for Gender Component", () => {
    expect(shallow(<Gender />)).toMatchSnapshot();
  });
});
