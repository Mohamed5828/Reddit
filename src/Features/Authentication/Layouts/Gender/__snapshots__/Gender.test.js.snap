// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gender Component this is a test for Gender Component 1`] = `
<Fragment>
  <styled.div>
    <styled.div>
      <styled.div>
        <span>
          <SiReddit
            color="red"
            size={60}
          />
        </span>
      </styled.div>
      <styled.span
        onClick={[Function]}
      >
        Skip
      </styled.span>
    </styled.div>
    <styled.h2>
      About you
    </styled.h2>
    <styled.div>
      Tell us about yourself to start building your home feed.
    </styled.div>
    <styled.div
      identity={true}
    >
      How do you identify?
    </styled.div>
    <GenderButton
      onClick={[Function]}
    >
      Woman
    </GenderButton>
    <GenderButton
      onClick={[Function]}
    >
      Man
    </GenderButton>
  </styled.div>
</Fragment>
`;
